src/kOS.Safe/Encapsulation/TerminalStruct.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe;
+
+namespace kOS.Safe.Encapsulation
+{
+    public class TerminalStruct : Structure
+    {
+        protected SharedObjects shared;
+        
+        // Some sanity values to prevent the terminal display from getting garbled up:
+        // They may have to change after experimentation.
+        protected const int MINROWS = 3;
+        protected const int MAXROWS = 80;
+        protected const int MINCOLUMNS = 15;
+        protected const int MAXCOLUMNS = 160;
+
+        // TODO: To implement IsOpen, we'd have to make a kOS.Safe interface wrapper around TermWindow first.
+        // That's more than I want to do in this update, I'm leaving it as a TODO for me or someone else:
+        //
+        // protected bool IsOpen { get { return shared.Window.IsOpen(); } set {if (value) shared.Window.Open(); else shared.Window.Close(); } }
```

Comment:
window will be pretty hard to move over on its own because it inherits from MonoBehavior. We could just give it an interface ITermWindow except for AttachTo that needs the kOS sharedobjects object.